pkg/services/dashboards/service/dashboard_service.go

```diff
@@ -1715,10 +1724,54 @@ func (dr *DashboardServiceImpl) searchDashboardsThroughK8sRaw(ctx context.Contex
 		request.Options.Fields = append(request.Options.Fields, req...)
 	}
 
+	if query.IsDeleted {
+		request.IsDeleted = query.IsDeleted
+	}
+
 	if query.Limit > 0 {
 		request.Limit = query.Limit
 	}
 
+	if query.Page > 0 {
+		request.Page = query.Page
+		request.Offset = query.Page
```

Comment:
We'll want to add a default limit [like in legacy](https://github.com/grafana/grafana/blob/main/pkg/services/dashboards/database/database.go#L968) if the page is set, otherwise if you send /api/search?page=2, it will return the same list as page=1 (which is different than the current behavior)

```suggestion
		request.Offset = query.Page
		if query.Limit == 0 {
			query.Limit = 1000
		}
```